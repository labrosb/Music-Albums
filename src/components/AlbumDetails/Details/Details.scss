@import '../../../styles/colors.scss';
@import '../../../styles/mixins.scss';

.album-details {
  width: 300px;
  margin: 0 16px;
  padding: 16px 30px;
  box-sizing: border-box;
  border: 0.5px dotted color(secondary);
  border-radius: 6px;
  position: relative;
  .details-row {
    display: flex;
    justify-content: space-between;
  }
  .light-cover {
    height: 100%;
    width: 100%;
    opacity: 0.1;
    z-index: -1;
    position: absolute;
    top: 0;
    left: 0;
  }
  .details-column {
    display: flex;
    flex-direction: column;
  }
  .album-artist {
    @include text(small, text-dark);
    margin-top: 6px;
  }
  .album-name {
    @include text(medium, text-dark);
    margin-top: 6px;
  }
  .album-detail {
    @include text(small, text-dark);
    margin-top: 6px;
  }
  .album-buttons {
    display: flex;
    justify-content: space-around;
    margin: 40px 0 30px 0;
    .album-button {
      @include text(small, background);
      display: flex;
      justify-content: center;
      align-items: center;
      padding: 7px 14px;
      background-color: color(text-dark);
      border-radius: 14px;
      cursor: pointer;
      .button-icon {
        margin-right: 8px;
      }
    }
    .album-button.favorite-button {
      background-color: color(favorite);
    }
  }
}

.album-price {
  @include text(tiny, text-dark);
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 16px;
  margin-left: 8px;
  border: 1px dashed color(primary);
  border-radius: 50%;
  height: 44px;
  min-width: 48px;
}

@media (max-width: 630px) {
  .album-details {
    width: 290px;
    margin: 20px 0;
  }
}
