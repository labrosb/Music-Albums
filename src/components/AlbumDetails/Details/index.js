import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart } from '@fortawesome/free-solid-svg-icons';
import './Details.scss';

function Details({ albumData, toggleFavorite, isFavorite }) {
  const onViewMoreClick = useCallback(() => {
    window.location.href = albumData.link;
  }, [albumData.link]);

  const onFavoriteClick = useCallback(() => {
    toggleFavorite(isFavorite, albumData);
  }, [albumData, isFavorite, toggleFavorite]);

  return (
    <div className="album-details">
      <div className="details-row">
        <div
          className="light-cover"
          style={{ backgroundImage: `url(${albumData.thumbnail})` }}
        />
        <div className="details-column">
          <div className="album-artist">
            {albumData.artist}
          </div>
          <div className="album-name">
            {albumData.name}
          </div>
          <br />
          <div className="album-detail">
            Music: {albumData.category}
          </div>
          <div className="album-detail">
            Int.Release {albumData.releaseDate}
          </div>
        </div>
        <div className="album-price">
          {albumData.price}
        </div>
      </div>
      <div className="album-buttons">
        <div
          className="album-button favorite-button"
          onClick={onFavoriteClick}
        >
          <FontAwesomeIcon
            className="button-icon"
            icon={faHeart}
          />
          <span>{isFavorite ? 'Remove' : 'Add'}</span>
        </div>
        <div
          className="album-button"
          onClick={onViewMoreClick}
        >
          View More
        </div>
      </div>
    </div>
  );
}

Details.propTypes = {
  albumData: PropTypes.object,
  isFavorite: PropTypes.bool,
  toggleFavorite: PropTypes.func
};

export default Details;
