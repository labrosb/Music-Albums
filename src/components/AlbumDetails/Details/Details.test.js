/* eslint-disable no-undef */
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Details from './index';

configure({ adapter: new Adapter() });

const albumData = {
  artist: 'Kenny Chesney',
  category: 'Country',
  id: '1502163294',
  link: 'https://music.apple.com/us/album/here-and-now/1502163294?uo=2',
  name: 'Here and Now',
  price: '$10.99',
  rank: 1,
  releaseDate: 'May 1, 2020',
  rights: 'â„— 2020 Blue Chair Records, LLC under exclusive license to Warner Music Nashville LLC.',
  thumbnail: 'https://is5-ssl.mzstatic.com/image/thumb/Music114/v4/3c/a4/90/3ca4901d-b744-411d-246e-02f78aebf8bb/093624892984.jpg/500x500bb.png'
};

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Details isFavorite {...{ albumData }} />);
});

describe('AlbumDetails -> Detais component', () => {
  it('Should render album name correctly', () => {
    const name = wrapper.find('.album-name').text();

    expect(name).toEqual(albumData.name);
  });

  it('Should render artist name correctly', () => {
    const artist = wrapper.find('.album-artist').text();

    expect(artist).toEqual(albumData.artist);
  });

  it('Should render album category correctly', () => {
    const category = wrapper.find('.album-detail').at(0).text();

    expect(category).toEqual(`Music: ${albumData.category}`);
  });

  it('Should render album release date correctly', () => {
    const date = wrapper.find('.album-detail').at(1).text();

    expect(date).toEqual(`Int.Release ${albumData.releaseDate}`);
  });

  it('Should render album price correctly', () => {
    const price = wrapper.find('.album-price').text();

    expect(price).toEqual(albumData.price);
  });

  it('Should render remove button if album is in favorites', () => {
    const buttonText = wrapper.find('.favorite-button').find('span').text();

    expect(buttonText).toEqual('Remove');
  });

  it('Should render add button if album is not in favorites', () => {
    wrapper.setProps({ isFavorite: false });
    const buttonText = wrapper.find('.favorite-button').find('span').text();

    expect(buttonText).toEqual('Add');
  });
});
