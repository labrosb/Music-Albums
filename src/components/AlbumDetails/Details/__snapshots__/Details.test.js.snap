// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewCard component Should match snapshot 1`] = `undefined`;
